controllers/templatesync/template_sync.go

```diff
@@ -522,26 +513,37 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		// Attempt to fetch the resource
 		eObject, err := res.Get(ctx, tName, metav1.GetOptions{})
 		if err != nil {
-			if len(dependencyFailures) > 0 {
-				// template must be pending, do not create it
-				emitErr := r.emitTemplatePending(ctx, instance, tIndex, tName, isClusterScoped,
-					generatePendingMsg(dependencyFailures))
-				if emitErr != nil {
-					resultError = emitErr
+			// not found should consider creating it
+			if k8serrors.IsNotFound(err) {
+				if len(dependencyFailures) > 0 {
```

Comment:
This section was moved inside the `IsNotFound` part: previously, it was possible that other errors would not be re-queued if the template was supposed to be Pending. Since those errors are rare, and this is already a complicated situation, I'm not sure if this ever caused any strange problems, but it's possible.